### PR TITLE
fix(accessibilty): user account page fixes #ACC-18

### PR DESCRIPTION
**Pull request by BartDyb**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-18" title="ACC-18" target="_blank">SubtaskACC-18</a>  9.2.1.1 Fix user account page elements
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Accessibility tabbing/focusing fixes for the user account page.

https://wearezeta.atlassian.net/browse/ACC-18

### CODE REVIEW

File: src/script/page/preferences/accountPreferences/AccountLink.tsx [hunk 1]
```diff
@@ -64,12 +64,13 @@ const AccountLink: React.FC<AccountLinkProps> = ({label, value, ...rest}) => {
           {value}
         </div>
       </div>
-      <div
+      <button
         data-uie-name="do-copy-profile-link"
```
Comment (kamil-au) [inline src/script/page/preferences/accountPreferences/AccountLink.tsx:69]: type is missing

File: src/script/components/AvailabilityState.tsx [hunk 2]
```diff
@@ -82,14 +108,30 @@ const AvailabilityState: React.FC<AvailabilityStateProps> = ({
...
+      {label && onClick && (
+        <button
+          className="availability-state-label"
+          css={
+            theme
+              ? {...buttonCommonStyles, color: 'var(--accent-color)', userSelect: 'none'}
+              : {...buttonCommonStyles, userSelect: 'none'}
+          }
+          data-uie-name="status-label"
+          onClick={onClick}
```
Comment (kamil-au) [inline src/script/components/AvailabilityState.tsx:131]: missing type

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx [hunk 3]
```diff
@@ -85,30 +88,40 @@ const AvatarInput: React.FC<AvatarInputProps> = ({selfUser, isActivatedAccount,
     }
   };
 
+  const handleKeyPress = (event: React.KeyboardEvent) => {
```
Comment (kamil-au) [inline src/script/page/preferences/accountPreferences/AvatarInput.tsx]: we have a `handleKeyDown` function as a wrapper in KeyboardUtils, maybe you can use it here?

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx [hunk 4]
```diff
@@ -85,30 +88,40 @@ const AvatarInput: React.FC<AvatarInputProps> = ({selfUser, isActivatedAccount,
...
-            const isInvalidUpdate = error.type === UserError.TYPE.INVALID_UPDATE;
-            if (!isInvalidUpdate) {
-              throw error;
-            }
-          });
-        }}
-      />
-      <span className="icon-camera"></span>
-    </label>
+    <div tabIndex={0} role="button" onKeyPress={handleKeyPress}>
```
Comment (kamil-au) [inline src/script/page/preferences/accountPreferences/AvatarInput.tsx]: is the screen reader reading the labels from the child when this one is focused or we should provide here an `aria-label`?

File: src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx [hunk 5]
```diff
@@ -24,57 +24,75 @@ import {HistoryExportViewModel} from '../../../view_model/content/HistoryExportV
...
-    <p id="preferences-history-describe-1" className="preferences-detail">
-      {t('preferencesOptionsBackupExportSecondary', brandName)}
-    </p>
-    <label
-      className="preferences-link accent-text preferences-history-backup-import-field"
-      data-uie-name="do-backup-import"
+const HistoryBackupSection: React.FC<HistoryBackupSectionProps> = ({brandName}) => {
+  const fileInputRef = React.useRef(null);
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (kamil-au) [inline src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx]: the same as above about keyboard utils

File: src/script/page/preferences/accountPreferences/AccountLink.tsx (hunk 1 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AccountLink.tsx:69]: 👍

File: src/script/components/AvailabilityState.tsx (hunk 2 above)
Comment (BartDyb) [reply src/script/components/AvailabilityState.tsx:131]: 👍

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx (hunk 3 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AvatarInput.tsx]: I'll check it

File: src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx (hunk 5 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx]: 👍

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx (hunk 4 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AvatarInput.tsx]: might be valid point ;)

File: src/script/components/AvailabilityState.tsx (hunk 2 above)
Comment (BartDyb) [reply src/script/components/AvailabilityState.tsx:131]: added

File: src/script/page/preferences/accountPreferences/AccountLink.tsx (hunk 1 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AccountLink.tsx:69]: added

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx (hunk 4 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AvatarInput.tsx]: added

File: src/script/page/preferences/accountPreferences/AvatarInput.tsx (hunk 3 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/AvatarInput.tsx]: done

File: src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx (hunk 5 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx]: done

File: src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx [hunk 6]
```diff
@@ -24,57 +24,78 @@ import {HistoryExportViewModel} from '../../../view_model/content/HistoryExportV
...
+      >
+        {t('preferencesOptionsBackupExportHeadline')}
+      </button>
+      <p id="preferences-history-describe-1" className="preferences-detail">
+        {t('preferencesOptionsBackupExportSecondary', brandName)}
+      </p>
+      <div
+        role="button"
+        tabIndex={0}
+        onKeyDown={(event: React.KeyboardEvent<HTMLElement>) => handleKeyDown(event, fileInputClick)}
```
Comment (kamil-au) [inline src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx:67]: Here we probably also can add `aria-label`l or `aria-labelledby`

**Review by kamil-au — APPROVED** [review]

File: src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx (hunk 6 above)
Comment (BartDyb) [reply src/script/page/preferences/accountPreferences/HistoryBackupSection.tsx:67]: done

**Pull request merged**